pkg/framework/deployment/deployment.go

```diff
@@ -0,0 +1,69 @@
+package deployment
+
+import (
+	"reflect"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+type DeploymentConfig struct {
+	K8sClient kubernetes.Interface
+	Logger    micrologger.Logger
+}
+
+type Deployment struct {
+	k8sClient kubernetes.Interface
+	logger    micrologger.Logger
+}
+
+func New(config DeploymentConfig) (*Deployment, error) {
+	if config.K8sClient == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.K8sClient must not be empty", config)
+	}
+	if config.Logger == nil {
+		return nil, microerror.Maskf(invalidConfigError, "%T.Logger must not be empty", config)
+	}
+
+	d := &Deployment{
+		k8sClient: config.K8sClient,
+		logger:    config.Logger,
+	}
+
+	return d, nil
+}
+
+// Check ensures that key properties of the deployment are correct.
+func (d *Deployment) Check(name string, replicas int, expectedLabels, expectedMatchLabels map[string]string) error {
+	ds, err := d.k8sClient.Apps().Deployments(metav1.NamespaceSystem).Get(name, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		return microerror.Newf("could not find deployment: '%s' %v", name, err)
```

Comment:
Eh, no. See https://github.com/giantswarm/fmt/blob/master/go/errors.md.

Reply:
Ahh :sweat_smile: too fast.

Reply:
The way to go here is to create a custom error matcher i.e. `IsNotFoundError` and wrap/match the apierror there, right? Similar to the promtheus example?

Reply:
Maybe. Depends on what you want to do. When the error here has to be managed in a special way then we should have control over it and return our own proper type.

Reply:
You could just use `microerror.Maskf` IMO. I am kind of wondering why there is a special case for `NotFound` to begin with. We don't do anything different with it, right?

Reply:
Decided to go for an own `notFoundError` type, so consuming packages can match it more easily. Also created and masked errors for the other cases. Is that done right?

Reply:
Do we want to act differently though? My question would be why not one `incorrectDeploymentError` to rule them all? Just thinking loud here.

Reply:
You have a good point `incorrectDeploymentError` is good to caputure all incorrect properties, be it labels or replicas. Still IMO `notFoundError` is semantically something different than `incorrect`, and can also cause different actions.